lib/compat/wordpress-6.8/blocks.php

```diff
@@ -20,77 +20,121 @@ function gutenberg_stabilize_experimental_block_supports( $args ) {
+		'typography' => array(
 			'__experimentalFontFamily'     => 'fontFamily',
 			'__experimentalFontStyle'      => 'fontStyle',
 			'__experimentalFontWeight'     => 'fontWeight',
 			'__experimentalLetterSpacing'  => 'letterSpacing',
 			'__experimentalTextDecoration' => 'textDecoration',
 			'__experimentalTextTransform'  => 'textTransform',
 		),
-		'__experimentalBorder' => 'border',
 	);
+	$done                            = array();
 
 	$updated_supports = array();
 	foreach ( $args['supports'] as $support => $config ) {
-		// Add the support's config as is when it's not in need of stabilization.
-		if ( empty( $experimental_to_stable_keys[ $support ] ) ) {
+		/*
+		 * If this support config has already been stabilized, skip it.
+		 * A stable support key occurring after an experimental key, gets
+		 * stabilized then so that the two configs can be merged effectively.
+		 */
+		if ( isset( $done[ $support ] ) ) {
+			continue;
+		}
+
+		$stable_support_key = $experimental_supports_map[ $support ] ?? $support;
+
+		/*
+		 * Use the support's config as is when it's not in need of stabilization.
+		 *
+		 * A support does not need stabilization if:
+		 * - The support key doesn't need stabilization AND
+		 * - Either:
+		 *     - The config isn't an array, so can't have experimental properties OR
+		 *     - The config is an array but has no experimental properties to stabilize.
+		 */
+		if ( $support === $stable_support_key &&
+			( ! is_array( $config ) ||
+				( ! isset( $experimental_support_properties[ $stable_support_key ] ) &&
+				empty( array_intersect_key( $common_experimental_properties, $config ) )
+				)
+			)
+		) {
 			$updated_supports[ $support ] = $config;
 			continue;
 		}
 
-		// Stabilize the support's key if needed e.g. __experimentalBorder => border.
-		if ( is_string( $experimental_to_stable_keys[ $support ] ) ) {
-			$stabilized_key = $experimental_to_stable_keys[ $support ];
+		$stabilize_config = function ( $unstable_config, $stable_support_key ) use ( $experimental_support_properties, $common_experimental_properties ) {
+			$stable_config = array();
+			foreach ( $unstable_config as $key => $value ) {
+				// Get stable key from support-specific map, common properties map, or keep original.
+				$stable_key = $experimental_support_properties[ $stable_support_key ][ $key ] ??
+							$common_experimental_properties[ $key ] ??
+							$key;
+
+				$stable_config[ $stable_key ] = $value;
 
-			// If there is no stabilized key present, use the experimental config as is.
-			if ( ! array_key_exists( $stabilized_key, $args['supports'] ) ) {
-				$updated_supports[ $stabilized_key ] = $config;
-				continue;
+				/*
+				 * The `__experimentalSkipSerialization` key needs to be kept until
+				 * WP 6.8 becomes the minimum supported version. This is due to the
+				 * core `wp_should_skip_block_supports_serialization` function only
+				 * checking for `__experimentalSkipSerialization` in earlier versions.
```

Comment:
That was the first option I looked into however it would require a tonne of updates across the block support PHP functions now and then again once the stabilization was in the core in the minimum supported version for Gutenberg.

This approach (leaving the experimental prop for two releases) only requires code in this 6.8 compat file, which will naturally be removed when 6.8 is the minimum version supported. No need for multiple rounds of updates.  

There's also the use case of 3rd party blocks that skip serialization checking `wp_should_skip_block_supports_serialization`.

Reply:
Hmm yeah that makes sense, best leave it as is.